### PR TITLE
Update Netwayste to use async/await

### PR DESCRIPTION
**Pull request by Manghi**
Related to #137 

- [x] Update cargo.toml to specify newer versions of futures and tokio
- [x] Rewrite `server` udp sink/stream interface
    - Refactored server set-up code for readability
    - Refactored packet transmission code to remove `Option` wrapping and use zero-length `Vector` instead.
- [x] Rewrite `client` udp sink/stream interface
- [x] Update `cli-client` to use new `client` changes
- [x] Update dissector to use `NetwaysteUdpCodec` for decoding packets
- [x] Use tokio's built-in DNS (see #121)
    - Though not directly related, we might as well do this too. Easy win.
- [x] Update server unit tests
- [x] Update client and cli-client unit tests
- [x] No new warnings

### CODE REVIEW

**Comment by Manghi** [comment]
@AaronM04 I've left the `client` and `cli-client` async/await updates as an exercise for you. I'll continue to work on the unit tests and the dns ticket in the mean time.

**Comment by Manghi** [comment]
Still need to update the netwayste/client interface!

File: netwayste/src/client.rs [hunk 1]
```diff
@@ -19,20 +19,21 @@
...
 use std::error::Error;
-use std::io;
-use std::iter;
 use std::net::SocketAddr;
 use std::process::exit;
 use std::time::Duration;
 use std::time::Instant;
 
-use futures::{future::ok, stream, sync::mpsc, Future, Sink, Stream};
+use futures as Fut;
```
Comment (AaronM04) [inline netwayste/src/client.rs:27]: Same here.

File: conwayste/src/network.rs [hunk 2]
```diff
@@ -20,30 +20,30 @@
 extern crate futures;
 extern crate ggez;
 extern crate netwayste;
-extern crate tokio_core;
+extern crate tokio;
 
-use futures::sync::mpsc as futures_channel;
+use futures as Fut;
```
[the remark this answers is not in the document]
Comment (Manghi) [reply conwayste/src/network.rs:25]: I think single letter aliasing could get confusing when interspersed with generics, which is why I went with the abbreviation instead.

File: conwayste/src/network.rs [hunk 3]
```diff
@@ -20,30 +20,30 @@
...
-use std::process;
-use std::sync::mpsc as std_channel;
-use std::thread;
-use std_channel::TryRecvError;
-
 pub struct ConwaysteNetWorker {
-    sender:   futures_channel::UnboundedSender<NetwaysteEvent>,
-    receiver: std_channel::Receiver<NetwaysteEvent>,
+    sender:   Fut::channel::mpsc::UnboundedSender<NetwaysteEvent>,
+    receiver: Fut::channel::mpsc::Receiver<NetwaysteEvent>,
```
[the remark this answers is not in the document]
Comment (Manghi) [reply conwayste/src/network.rs:32]: I don't anticipate any issues with this.

**Pull request merged**